### DISCUSSION TITLE
Exit Code 1 - Unable to Download Video

### DISCUSSION THREAD

**Discussion by clubmmm** [aclap-dev/video-downloadhelper#290] [Help and Q&A]
### Your question

I am trying to download the first squarespace hosted video stream from https://www.celesteforutah.com and I am getting an error, exit code 1 and am unable to download. I am a new user and just now installed the chrome extension, installed the CoApp and paid for a new license. Can anybody please help me?

Error: Exit code: 1
    at rs.receive (chrome-extension://lmjnegcaeklhafolokijcfjliaokphfk/background/main.js:1:13157)
    at chrome-extension://lmjnegcaeklhafolokijcfjliaokphfk/background/main.js:1:42624

Note: I made sure that I do not have a firewall, defender, or other filter blocking anything.
Note: I also just tried the BETA chrome extension and am seeing the same thing.
Note: Looked at testing on Firefox, but did not want to pay an additional $28 as I already paid a license fee and was disappointed that the same license would not work on firefox. Would this work better on Firefox? It is not a youtube video, but a square space hosted video stream. 

### Addon & CoApp version and details

The "Copy information to clipboard" button added nothing to my clipboard, but here is the information for my extension and CoApp version:

- Version 8.2.1.1 (stable)
- Platform Win x86-64 Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/122.0.0.0 Safari/537.36
- Found companion app: VdhCoApp 2.0.19
- License Verified

![Video DownloadHelper Screenshot](https://github.com/aclap-dev/video-downloadhelper/assets/144069534/c314f6b6-d754-4fd4-a4b8-cfe48629d50d)

**Comment by GrampaWildWilly** [comment]
I am running Windows 11 Pro 64-bit version 23H2 build 22631.3296, Firefox 123.0.1 64-bit, licensed VDH 8.2.2.7a1 beta, CoApp 2.0.19.  I do not work for VDH.  I am just another user.

I visited your page & this is what I saw:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/dbbaeac6-59c1-45f3-9c38-0f2e57328da3)

Since the VDH menu was long, I scrolled it to display the rest of it.  It took 2 scrolls:

![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/67818ca5-99b5-424c-8aea-10d308314569)

My attempt to download the 1920x1080 mp4 of duration 0:25 failed with the error you reported.  My download of the 3840x2160 mkv of duration 1:41 went fine.

![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/5cc70c68-d246-4372-b728-d6ab3ffb5771)

Here's my results:

![#04](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/1c144502-fc1d-46bf-ac0d-1976b2c19906)

Here it is playing in VLC:

![#05](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/dff36d85-6372-41f5-b881-561daae6abc8)

You can see it's playing fine.  You'll have to take my word for it that the audio was just as fine.  I didn't sit & watch this, just sampled it enough to verify that it was all there.

I guess some content on that site is accessible by VDH & some isn't.  I'm guessing that's just the way it goes.  Maybe @paulrouget would have something to add.  Unlike me, he actually does work for VDH.  He's the developer.

**Reply by clubmmm** [reply]
Thank you for your testing... that's nice of you.

I was hoping to download the first square-space/hosted/streamed video (Celeste for congress and NOT the youtube video you were able to do).

Any ideas why the video I am trying to get is not working?

**Reply by GrampaWildWilly** [reply]
I was going to say no, I got the same error you did.  But I decided to look deeper.  I displayed the Hit Details for the duration 0:25 video.  There was the URL of an HLS master manifest shown there.  So I pumped it through ffprobe:

[ffprobe #1.txt](https://github.com/aclap-dev/video-downloadhelper/files/14608968/ffprobe.1.txt)

Well, when ffprobe (or ffmpeg) gives you an error message, you should usually pay attention to it.  I accepted the advice offered there & added `-allowed_extensions ALL` to my invocation of ffprobe.  Magic . . .

[ffprobe #2.txt](https://github.com/aclap-dev/video-downloadhelper/files/14608975/ffprobe.2.txt)

So this gave me what I needed to download this with ffmpeg:

[Political Video mp4 Log.txt](https://github.com/aclap-dev/video-downloadhelper/files/14608977/Political.Video.mp4.Log.txt)

Look carefully in that log.  You will see the exact ffmpeg & ffprobe commands I executed.

My results looked like this:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/3db38c08-dc87-4d12-bea0-3d616d646b3d)

And here it is playing just fine:

![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/dbeb557b-abd6-4489-a262-27d7b6d9e91b)

I have now added `-allowed_extensions ALL` to my habitual invocations of ffprobe & ffmpeg.  It seems that they are picky about file names.  I say that's a poor excuse to reject a download.  As you can see, things went fine for me.  I am expecting @paulrouget will come to the same conclusion & adjust the invocations of ffprobe & ffmpeg within . . .  Honestly, I don't know where in the product the invocations of ff* are.  I am guessing they're inside the CoApp.  But like I always say, I don't work for VDH so I don't really know.  Yes, the CoApp is open source & I could make changes there myself.  But that would require knowledge of the programming language of the CoApp, something I do not have.  I'm a programmer, but I don't know every programming language.  I should think it will be as easy for Paul to make this change in VDH/CoApp as it was for me to do in my stuff.

**Reply by GrampaWildWilly** [reply]
About `-allowed_extensions`.  I've encountered an error that made me look the parameter up in the usually useless ffmpeg documentation.  For a change, I actually found something relevant.  It turns out `-allowed_extensions` is valid as a parameter only when you're dealing with HLS.  So you can't download a simple file or a DASH stream & code that parameter.  On the other hand, you can code it on all invocations of ffprobe regardless of the file type in question.

**Reply by Mattmjs** [reply]
Is this another case to try to disrupt downloading, sort of like those missing audio from video.
The stable extension was not getting the audio but on the beta it did. That must mean ffmpeg is receiving instructions on
what to do from the extension. Maybe like your adjustment of the bat file you use.

**Reply by GrampaWildWilly** [reply]
You got something from both the general availability as well as the beta from that 0:25 duration variant?  I got an error & so did our original poster.  No audio, no video.  Nothing but an error.  He had it with the general availability VDH on Chrome.  I had it with the beta on Firefox.  If you managed to avoid an error & download anything at all, that's more than we managed to do.

I don't know that this is anything intentional or malicious on the part of the web site.  I don't think they are trying to thwart downloaders.  I think they're just a bit careless about following conventions.  They've got files with weird names.  The files are otherwise correct.  It's just that the file names are off.

**Reply by clubmmm** [reply]
Thanks @GrampaWildWilly!!!

My newb restatement for understanding: The VDH extension and it's CoApp use ffmpeg under the hood to download and convert the video streams from the website of interest. It was encountering an error because of unconventional naming. You used ffprobe to understand the stream definition and url's (m3u8). Then you used ffmpeg to stream and convert the video to a single *.mp4 video file.

In summary: VHD currently does not support this stream and file naming scenario, but using ffmpeg directly you can convert the stream to mp4. You gave a nice pointer for the VHD devs to make a change in future releases that would likely fix/support scenarios like this.

Thank you so much for your help! I tried this and it worked for me and now I also have a better understanding of how VHD functions. This workaround is fine for me as I only have a few of these scenarios that I need to scrape (for now). Thanks again!

**Comment by Mattmjs** [comment]
No, I was just making a comparison to
discussion #279 that a little something extra is required in ffmpeg in
order to
make it work.

You got something from both the general availability as well as the beta
> from that 0:25 duration variant?
>
